### ISSUE TITLE
[Ask] Trying to parse a response and the proxy just stops

### ISSUE THREAD

**Issue by alexandresgf** [abhinavsingh/proxy.py#1101]
Hello,

I'm trying to parse the response from `handle_upstream_data` to check its status code. The thing is, when I try to do it, the application just stops to work and nothing happens. Is there anything that I miss?

Here is my code:

```
def handle_upstream_data(self, raw: memoryview) -> None:
        logger.debug("Creating the Response Parser...")
        response = HttpParser(httpParserTypes.RESPONSE_PARSER)
        logger.debug("Parsing the upstream data...")
        response.parse(raw) # It runs forever
        logger.debug(
            "HTTP Response status code: %d",
            int.from_bytes(response.code),
        )
        super().handle_upstream_data(raw)
```
Just the debug logs below...

![Seleção_021](https://user-images.githubusercontent.com/4246277/156863645-e56c1cf1-31a2-4c4b-9ef8-fe7ccae67e2d.png)

**Comment by abhinavsingh** [comment]
@alexandresgf Can you share the contents of `raw`.  Ideally, you must check for `parser.state` before considering that complete request has been received.  But if `.parse` is running into in infinite loop, that smells like a bug.  Can you please share the raw packet.  Thank you!!!

**Comment by alexandresgf** [comment]
This is the output of `raw` parameter inside the `handle_upstream_data`:

```
HTTP/1.1 200 OK
x-luminati-ip: <IP> # I just suppressed it
x-luminati-timeline: z10977-init:0,auth:12,dns_resolve_skip:0,ipc_find:10,ext_conn:1,zlum_ztun_conn:0,ztun_conn:89
```

This output above appears about 3 times during my tests, then it just stops working.

To print it I used:

```
logger.debug(raw.tobytes().decode())
```

Some images below:

1. Trying to httpbin

![Seleção_022](https://user-images.githubusercontent.com/4246277/157119265-e3c89ef7-c880-40eb-927d-e1efde5bbde0.png)

2. After the method beign called for the 3rd time (I counted it), it just stops

![image](https://user-images.githubusercontent.com/4246277/157119506-86b17dde-8133-4ad2-a24d-fbc4670840e9.png)

I tested it many times with distincts proxies and I checked for `response.is_complete` and it never changes to `True`.
